### ISSUE TITLE
[feature request] Add CloudFormation Outputs to parameter store

### ISSUE THREAD

**Issue by bpottier** [aws/copilot-cli#4160]
We use AWS CDK to build some resources for our applications that we feel are better done outside the addons feature of copilot. 

I'm wondering if it would be useful (beyond our own use case) to create a parameter with all the values for any outputs created by copilot stacks. 
The reason being: in some instances of CDK we can't rely on doing a cloudformation import because it remains a token (and not a concrete value) until deployment. This prevents us from doing something like:

```
    const vpc = ec2.Vpc.fromLookup(this, 'VPC', {
      isDefault: false,
      vpcId: Fn.importValue("<copilot application>-<copilot environment>-VpcId")
    });
```

Maybe there is a better approach? Just wanted to bring up the idea.

**Comment by dannyrandall** [comment]
Hey @bpottier! Am I understanding this right?

You have some resources that you deploy using CDK. You want to reference resources created by Copilot from the CDK (like importing the VPC created by Copilot). The problem is that you can't deploy your CDK resources until _after_ the Copilot resources are created, because `Fn.importValue` doesn't work until you've run `copilot svc/env deploy` and the outputs exist in CloudFormation.

If that's correct ^, my question is around what you'd expect from the parameter alternative - what would happen in your CDK code if the parameter hasn't been created yet (like if you haven't run `copilot env deploy` yet)? And how is that different from what happens with `Fn.importValue` today?

---
As a side note - we are thinking a lot right now about how to integrate CDK with Copilot to bring the power of CDK to extend resources created by Copilot. If you don't mind sharing some more of your CDK use case, that would be super helpful!
- What types of resources are you creating using the CDK? 
- What "level" do those resources exist on - service, environment, or application?
- Is the lifecycle of those resources tied to any of those levels (i.e., a VPC Endpoint that gets created/deleted with an environment)

**Comment by bpottier** [comment]
It isn’t that we can’t deploy our CDK resources until after copilot deploy (although that is true because we do rely on copilot to provision our VPC at the moment) but we think there is a better solution for referencing copilot resources other than looking up arns etc and writing them into our CDK stack. 

What I mean is, we’d like to reference resources created by copilot in a more abstracted way, using the application name and environment name etc. 

The example I gave was relying on application name and environment name to import the vpc id. However, it doesn’t work for a VPC lookup because it requires a concrete value and not a [token](https://docs.aws.amazon.com/cdk/v2/guide/tokens.html), which is why I was suggesting writing exported values to something like the parameter store. 

Our main use case is creating service specific resources like RDS databases, elastic cache clusters. Frankly, we hate writing cloudformation but love CDK. But we still prefer to use copilot for the management of our containerized workloads because it’s a tool that solves a lot of things we don’t want to solve ourselves. 

Does that help? I’d be happy to answer any more questions.

**Comment by efekarakus** [comment]
Hi @bpottier !

We just posted a proposal for extending Copilot with the CDK: https://github.com/aws/copilot-cli/issues/4208, we would love your feedback ❤️

**Comment by benjaminpottier** [comment]
> Hi @bpottier !
> 
> We just posted a proposal for extending Copilot with the CDK: #4208, we would love your feedback ❤️

Absolutely! Might be a little bit. Thanksgiving this week and then re:Invent.
